src/lib.rs

```diff
@@ -44,14 +44,73 @@ fn unsafe_alloc_buffer(len: usize) -> Vec<u8> {
     ret
 }
 
+#[derive(Copy, Clone)]
+pub struct Uint128 {
+    pub bytes: [u8; 16],
+}
+
+#[derive(Copy, Clone)]
+pub struct Uint160 {
+    pub bytes: [u8; 20],
+}
+
+#[derive(Copy, Clone)]
+pub struct Uint256 {
+    pub bytes: [u8; 32],
+}
+
+type EtherValue = Uint128;
+type Address = Uint160;
+type StorageKey = Uint256;
+type StorageValue = Uint256;
+type Hash = Uint256;
```

Comment:
Only `EtherValue` should be uint, the rest is "hash". The difference is the byte order. EtherValue supposed to be little endian, the rest just bytes read from a location (e.g. "big endian" if you would want to read it as a number).